src/transformers/models/vit_mae/feature_extraction_vit_mae.py

```diff
@@ -0,0 +1,150 @@
+# coding=utf-8
+# Copyright 2021 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Feature extractor class for ViT MAE."""
+
+from typing import Optional, Union
+
+import numpy as np
+from PIL import Image
+
+from ...feature_extraction_utils import BatchFeature, FeatureExtractionMixin
+from ...file_utils import TensorType
+from ...image_utils import (
+    IMAGENET_DEFAULT_MEAN,
+    IMAGENET_DEFAULT_STD,
+    ImageFeatureExtractionMixin,
+    ImageInput,
+    is_torch_tensor,
+)
+from ...utils import logging
+
+
+logger = logging.get_logger(__name__)
+
+
+class ViTMAEFeatureExtractor(FeatureExtractionMixin, ImageFeatureExtractionMixin):
```

Comment:
It looks exactly like `ViTFeatureExtractor`. If there are no differences, the `ViTFeatureExtractor` should be reused instead of creating a new class (and you can map `ViTMAEConfig` to `ViTFeatureExtractor` in the auto mapping).

Reply:
It is identical, except that it uses the ImageNet default mean and std instead of the standard ones.

Will this still work? Should I just upload the `preprocessor_config.json` to the models on the hub, to make sure they use the appropriate mean and std?

Reply:
Yes, with the right preprocessor config, it should work.